test/integration/copy_paste_spec.js

```diff
@@ -197,4 +198,116 @@ describe("Copy and paste", () => {
       );
     });
   });
+
+  describe("Copy/paste and ligatures (but without selecting all)", () => {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait(
+        "copy_paste_ligatures.pdf",
+        "#hiddenCopyElement",
+        100,
+        async page => {
+          await page.waitForFunction(
+            () => !!window.PDFViewerApplication.eventBus
+          );
+          await waitForTextLayer(page);
+        }
+      );
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    it("must check that the ligatures have been removed when the text has been copied", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          await page.evaluate(async () => {
+            const promise = new Promise(resolve => {
+              document.addEventListener("selectionchange", resolve, {
+                once: true,
+              });
+            });
+            const range = document.createRange();
+            range.selectNodeContents(document.querySelector(".textLayer"));
+            const selection = window.getSelection();
+            selection.addRange(range);
+            await promise;
+          });
+          const promise = page.evaluate(
+            () =>
+              new Promise(resolve => {
+                document.addEventListener(
+                  "copy",
+                  e => resolve(e.clipboardData.getData("text/plain")),
+                  {
+                    once: true,
+                  }
+                );
+              })
+          );
+          await page.keyboard.down("Control");
+          await page.keyboard.press("c");
+          await page.keyboard.up("Control");
+          const text = await promise;
+
+          if (browserName === "chrome") {
+            // In Firefox the copy event hasn't the correct value.
```

Comment:
Can you please add a TODO with a link to a relevant upstream bug-report, either in Firefox or Puppeteer (whichever is appropriate), so that we can eventually enable this test everywhere?